test/specs/core.datasetController.tests.js

```diff
@@ -38,6 +38,80 @@ describe('Chart.DatasetController', function() {
 		});
 	});
 
+	describe('inextensible data', function() {
+		it('should handle a frozen data object', function() {
+			function createChart() {
+				var data = Object.freeze([0, 1, 2, 3, 4, 5]);
+				var chart = acquireChart({
+					type: 'line',
+					data: {
+						datasets: [{
+							data: data
+						}]
+					}
+				});
+	
+				chart.data.datasets[0].data = Object.freeze([5, 4, 3, 2, 1, 0]);
+				chart.update();
+	
+				// Tests that the unlisten path also works for frozen objects
+				chart.destroy();
+			}
+	
+			expect(createChart).not.toThrow();
+		});
+
+		it('should handle a sealed data object', function() {
+			function createChart() {
+				var data = [0, 1, 2, 3, 4, 5];
```

Comment:
Since you need to update this file (Travis is failing on linting), you could change this line for `var data = Object.seal([0, 1, 2, 3, 4, 5]);` as in the first test? Same for following lines.

Reply:
Done. It wasn't clear from the MDN docs that it worked liked that.

Reply:
Good point (actually, I didn't check if it was working :) ). What do you think to add a sanity check to make sure these tests are correctly checking the expected use case, for example:

```javascript
var data = Object.freeze([0, 1, 2, 3, 4, 5]);
expect(Object.isExtensible(data)).toBeFalsy();

var chart = acquireChart({ 
    // ...
});

var dataset = chart.data.datasets[0];
dataset.data = Object.freeze([5, 4, 3, 2, 1, 0]);
expect(Object.isExtensible(dataset.data)).toBeFalsy();

chart.update();
```

Reply:
I had tested manually before making changes. Good point on testing the setup though. I went ahead and changed that.